source/agora/utils/InetUtils.d

```diff
@@ -139,10 +160,97 @@ struct InetUtils
 
                 adapter_info = adapter_info.Next;
             }
+            ips ~= publicAddresses();
             return ips;
         }
     }
 
+    static string publicAddressImpl (Service service, AddressFamily addressFamily)
+        @safe
+    {
+
+        Address address =
+        {
+            switch (addressFamily) 
+            {
+                case AddressFamily.INET: return cast(Address)new InternetAddress(service.host, 80);
+                case AddressFamily.INET6: return cast(Address)new Internet6Address(service.host, 80);
+                default: throw new SocketException("Invalid address family");
+            }
+        }();
+
+        Socket socket = new TcpSocket(addressFamily);
+        socket.blocking = true;
+        socket.setOption(SocketOptionLevel.SOCKET, SocketOption.SNDTIMEO, dur!"seconds"(5));
+        socket.setOption(SocketOptionLevel.SOCKET, SocketOption.SNDTIMEO, dur!"seconds"(5));
+        socket.connect(address);
+        scope(exit) socket.close();
+
+        if (socket.send("GET " ~ service.path ~ " HTTP/1.1\r\nHost: " ~ service.host ~ "\r\nAccept: text/plain\r\n\r\n") 
+            != Socket.ERROR)
+        {
+            char[] buffer = new char[512];
+            ptrdiff_t recv, body_;
+
+            if ((recv = socket.receive(buffer)) != Socket.ERROR && (body_ = buffer[0..recv].indexOf("\r\n\r\n")) != -1)
+                return buffer[body_+4..recv].idup.strip;
+        }
+
+        throw new SocketException("Could not send or receive data");
```

Comment:
Why are we sending raw HTTP requests instead of using Vibe.d ? Especially considering those are blocking.

Reply:
OK, I agree that Non-blocking is right for us and we should use Vibe.d.